test/join.jl

```diff
@@ -887,4 +887,16 @@ end
     @test_throws ArgumentError join(df1, df2, on=:id, kind=:inner)
 end
 
+@testset "Make DataFrameJoiner rely on AbstractDataFrame" begin
```

Comment:
```suggestion
@testset "join mixing DataFrame and SubDataFrame" begin
```
(Would it make sense to test other joins?)

Reply:
no - it is OK to test only this